src/parquet/file/writer-internal.cc

```diff
@@ -142,7 +142,7 @@ ColumnWriter* RowGroupSerializer::NextColumn() {
   // Throws an error if more columns are being written
   auto col_meta = metadata_->NextColumnChunk();
 
-  if (current_column_writer_) { total_bytes_written_ = current_column_writer_->Close(); }
+  if (current_column_writer_) { total_bytes_written_ += current_column_writer_->Close(); }
```

Comment:
The fix you made here is correct. However, I don't think we are using `total_bytes_written_` to set the `total_byte_size` value in the row group metadata. We are re-computing it in the metadata writer over all the columns written.
https://github.com/apache/parquet-cpp/blob/master/src/parquet/file/metadata.cc#L471
 `total_byte_size += row_group_->columns[i].meta_data.total_compressed_size;`
 `................. `
`row_group_->__set_total_byte_size(total_byte_size);`